org.openscad.OpenSCAD.json

```diff
@@ -123,50 +133,98 @@
+                "--without-magick-plus-plus",
+                "--without-openexr",
+                "--without-openjp2",
+                "--without-pango",
+                "--without-raqm",
+                "--without-tiff",
+                "--without-webp",
+                "--without-wmf",
+                "--without-x",
+                "--without-xml",
+                "--without-zlib"
+            ],
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://github.com/ImageMagick/ImageMagick/archive/7.0.9-6.tar.gz",
+                    "sha256": "83517787920fc3a1215baf2b8025fdc6f2a09a934fd59b5007eaa214225ca5fc"
+                }
+            ]
+        },
         {
             "name": "openscad",
             "buildsystem": "qmake",
             "config-opts": [
                 "QMAKE_CFLAGS_ISYSTEM=",
-                "QMAKE_LIBDIR=/app/lib"
+                "QMAKE_LIBDIR=/app/lib",
+                "QMAKE_CXXFLAGS+=-fext-numeric-literals"
+            ],
+            "cleanup": [
+                "/share/pixmaps"
             ],
             "sources": [
                 {
                     "type": "archive",
-                    "url": "http://files.openscad.org/openscad-2015.03-2.src.tar.gz",
-                    "sha256": "a2535bb9e27d96e10e1e19268a596155164c7129d410a0c7f96edc2d09400083"
-                },
-                {
-                    "type": "patch",
-                    "path": "UIC-gettext-interoperability.patch"
-                },
-                {
-                    "type": "patch",
-                    "path": "openscad-libscintilla.patch"
-                },
-                {
-                    "type": "patch",
-                    "path": "openscad-appdata-update-1.patch"
-                },
-                {
-                    "type": "patch",
-                    "path": "openscad-appdata-update-2.patch"
+                    "url": "https://files.openscad.org/openscad-2019.05.src.tar.gz",
+                    "sha256": "0a16c4263ce52380819dd91c609a719d38f12f6b8c4da0e828dcbe5b70996f59"
                 }
             ],
             "post-install": [
-                "mkdir -p /app/share/icons/hicolor/512x512/apps",
-                "mv /app/share/pixmaps/openscad.png /app/share/icons/hicolor/512x512/apps",
                 "mv /app/share/mime/packages/openscad.xml /app/share/mime/packages/org.openscad.OpenSCAD.xml",
-                "sed -i -e 's|<icon name=\"openscad\"/>|<icon name=\"org.openscad.OpenSCAD\"/>|' /app/share/mime/packages/org.openscad.OpenSCAD.xml"
+                "sed -i -e 's|<icon name=\"openscad\"/>|<icon name=\"org.openscad.OpenSCAD\"/>|' /app/share/mime/packages/org.openscad.OpenSCAD.xml",
+                "sed -i -e '/<releases>/a\\    <release date=\"2019-05-13\" version=\"2019.05\"/>' /app/share/metainfo/org.openscad.OpenSCAD.appdata.xml",
+                "for s in {16,32,48,64,128,256,512}; do
```

Comment:
I don't think this is really necessary.
As long as a large size is installed, toolkits should be able to size the icons down dynamically as needed. Also you can drop the imagemagick build dependency if you remove this.

Reply:
It's a requirement of Flathub: https://github.com/flathub/flathub/wiki/App-Requirements#application-icons

Reply:
This will go away with the next update of OpenSCAD via https://github.com/openscad/openscad/pull/3094 by installing 48x48 and 256x256 icons. Other resolutions should be available though scaling. The current problem is that the installed file is non-standard 688 x 688 size causing flathub build to fail.

Reply:
Oh, just reading that link, flatpak seems to require 64x64 and 128x128?

Reply:
> flatpak seems to require 64x64 and 128x128?

It's a requirement of Flathub. Flatpak itself doesn't care.

Reply:
Added upstream via https://github.com/openscad/openscad/pull/3173.

Reply:
I would recommend including 512x512 at the very least, ideally 256x256 as well. With hidpi becoming more common, 128x128 is arguably insufficient in many contexts, and this gives DEs more options.

Reply:
Ok, why not, added 512x512 too. That now means we have everything between 48 and 512.